### PR TITLE
Added initial replication settings to domain settings

### PR DESCRIPTION
**Pull request by seefo**
Addresses first part of https://highfidelity.fogbugz.com/f/cases/5284/Add-setting-group-Replication-to-domain-server-settings

No test plan yet

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/594/HighFidelity-Beta-PR10667-8b150547e990541da107e977ff9f370bd6688a28.dmg

File: domain-server/resources/describe-settings.json [hunk 1]
```diff
@@ -1318,6 +1318,53 @@
...
+            {
+              "name": "user",
+              "label": "User",
+              "can_set": true
+            }
+          ]
+        },
+        {
+          "name": "replicants",
+          "label": "Replicants",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: Can we make this name `downstream_servers` and label `Downstream Servers`

File: domain-server/resources/describe-settings.json [hunk 2]
```diff
@@ -1318,6 +1318,53 @@
...
+          "type": "table",
+          "can_add_new_rows": true,
+          "help": "Nodes or mirrors that are to replicate the domain",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: Change this to "Downstream servers that are relayed data for replicated users"

File: domain-server/resources/describe-settings.json [hunk 3]
```diff
@@ -1318,6 +1318,53 @@
...
+    {
+      "name": "replication",
+      "label": "Replication",
+      "settings": [
+        {
+          "name": "users",
+          "label": "Replicated Users",
+          "type": "table",
+          "can_add_new_rows": true,
+          "help": "Users that are replicated",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: "Users that are replicated to downstream servers"

File: domain-server/resources/describe-settings.json [hunk 4]
```diff
@@ -1318,6 +1318,53 @@
...
+          "columns": [
+            {
+              "name": "user",
+              "label": "User",
+              "can_set": true
+            }
+          ]
+        },
+        {
+          "name": "replicants",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: add `assignment-types` for this setting and make it `[0,1]`

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/604/HighFidelity-Beta-PR10667-587ab28d6c7eef02f825d3f8778ef4fdc3007f77.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/604/HighFidelity-Beta-PR10667-587ab28d6c7eef02f825d3f8778ef4fdc3007f77.dmg

File: domain-server/resources/describe-settings.json [hunk 5]
```diff
@@ -1318,6 +1318,54 @@
           "advanced": true
         }
       ]
+    },
+    {
+      "name": "replication",
+      "label": "Replication",
+      "assignment-types": [0,1],
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: move this down to the `downstream_servers` setting only, not here

File: domain-server/resources/describe-settings.json [hunk 6]
```diff
@@ -1318,6 +1318,54 @@
...
+              "label": "Address",
+              "can_set": true
+            },
+            {
+              "name": "port",
+              "label": "Port",
+              "can_set": true
+            },
+            {
+              "name": "mixer_type",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: this should be server_type

File: domain-server/resources/describe-settings.json [hunk 7]
```diff
@@ -1318,6 +1318,54 @@
...
+              "label": "Mixer Type",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: This should be Server Type

File: domain-server/resources/describe-settings.json [hunk 8]
```diff
@@ -1318,6 +1318,54 @@
...
+              "placeholder": "avatar-mixer or audio-mixer",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: make this "Audio Mixer"

File: domain-server/resources/describe-settings.json [hunk 9]
```diff
@@ -1318,6 +1318,54 @@
...
+            {
+              "name": "user",
+              "label": "User",
+              "can_set": true
+            }
+          ]
+        },
+        {
+          "name": "downstream_servers",
+          "label": "Replicants",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: label should be Downstream Servers here

File: domain-server/resources/describe-settings.json [hunk 10]
```diff
@@ -1318,6 +1318,54 @@
...
+          "assignment-types": [0,1],
+          "type": "table",
+          "can_add_new_rows": true,
+          "help": "Downstream Servers",
```
Comment (birarda) [inline domain-server/resources/describe-settings.json]: "Downstream servers that are relayed data for replicated users"

**Comment by birarda** [comment]
Merging this without waiting on a build because it is a JSON only change

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/617/HighFidelity-Beta-PR10667-6635187da36253048be2c086352ee82a87bc602d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/617/HighFidelity-Beta-PR10667-6635187da36253048be2c086352ee82a87bc602d.exe

**Pull request merged**